Engine/Core/Loader.cpp

```diff
@@ -867,6 +893,12 @@ int main(int argc, char **argv) {
 	// }
 	// Deinitialisation is done automatically by ctrl.quit(exit_code);
 
+#if defined(DISCORD)
+	if (opts.find("discord") != opts.end()) {
+		shutdownDiscord();
```

Comment:
This will not be called in many cases. I believe you should move the discord code to controllers really.